Sources/Workspace/Workspace.swift

```diff
@@ -2567,10 +2623,34 @@ public final class LoadableResult<Value> {
 private struct RemoteArtifact {
     let packageRef: PackageReference
     let targetName: String
-    let url: String
+    let url: Foundation.URL
     let checksum: String
 }
 
+private struct ArchiveIndexFile: Decodable {
+    let schemaVersion: String
+    let archives: [Archive]
+
+    struct Archive: Decodable {
+        let fileName: String
```

Comment:
Should we eventually support a subpath here?  Perhaps it isn't important.

Reply:
I was hoping to simplify by assuming a flat structure, but open to sub-directory if you think it helps

Reply:
No, we can stay with a file name for now — it's straightforward to extend in the future if we need to.